### PR TITLE
Remove OAuth 1 ApiCapability user branch

### PR DESCRIPTION
**Pull request by AntonKhorev**
I looked at Coveralls and noticed that `token.user` is never accessed:

image

I guess this branch is not required after #5058.

### CODE REVIEW

File: app/abilities/api_capability.rb [hunk 1]
```diff
@@ -5,11 +5,7 @@ class ApiCapability
 
   def initialize(token)
     if Settings.status != "database_offline"
-      user = if token.respond_to?(:resource_owner_id)
-               User.find(token.resource_owner_id)
-             elsif token.respond_to?(:user)
-               token.user
-             end
+      user = (User.find(token.resource_owner_id) if token.respond_to?(:resource_owner_id))
```
Comment (tomhughes) [inline app/abilities/api_capability.rb:8]: We don't actually need the condition at all - we should always have a token and it should always be an OAuth 2 token now.

There is a test that will need fixing but it's a bogus test that deliberately passes `nil` as the token which is not valid. The real code never does that.

**Review by tomhughes — CHANGES_REQUESTED** [review]

File: app/abilities/api_capability.rb (hunk 1 above)
Comment (AntonKhorev) [reply app/abilities/api_capability.rb]: Removed that part of the test because you don't have any abilities by default and it only tested `cannot?`.

**Review by tomhughes — APPROVED** [review]
Thanks - that looks good to me now.
